### ISSUE TITLE
Proposal: FortuneBlock: A Decentralized Lottery Game on Kairos

### ISSUE THREAD

**Issue by Spydiecy** [kaiachain/kaia-dapp-mono#266]
**Description:**
A transparent lottery game featuring:

Daily, weekly, and monthly draws with KAIA token prizes
Secure random winner selection
Instant automated payouts
User profiles and lottery history tracking

**Bounty track:** 
others, Gamefi

**Comment by zxstim** [comment]
Well automated payout is not possible with evm. What I would do is making a function that lets anyone to trigger the payout mechanism for 5% + gas fee of the payout amount; or let the winner to claim the amount. 

This way people would race to do this for you without the need for you to set up a cron yourself (because your cron might go down, your server goes down and no one gets paid, so the idea is having backup)

**Comment by Spydiecy** [comment]
It is all done with evm though i wrote a code so that as soon as the results of a lottery come in the total prizepool received on the lottery in the contract are sent to the random winner selected

**Comment by Spydiecy** [comment]
like i have set a timer for a lottery duration
after that duration ends the funds transfer happens

**Comment by zxstim** [comment]
well basically someone else is triggering the payout for you. evm requires a tx to be submitted.

what i referred to is that there is no "long running" contract that would do payout automatically like with other VM.

in this case i think you are referring to an oracle fulfilling call?

**Comment by Spydiecy** [comment]
Yes sort of like an oracle, like the payout would be directly done by that mechanism as soon as i have the private key and balance in the .env with a serverish frontend. Also pls check this out, have submitted everything in it: https://github.com/kaiachain/kaia-dapp-mono/issues/259

**Comment by Spydiecy** [comment]
Could work with the manual thing too which you recommended
